src/Metadata/StorageBase.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Tuf\Metadata;
+
+/**
+ * Defines a base class for saving and loading trusted TUF metadata.
+ */
+abstract class StorageBase implements StorageInterface
```

Comment:
Do we want to mark this as `@internal`? Since I guess the interface would be the new API?

Reply:
We could, but since this is a base class I would expect people to extend this. So to me, this is API as well.

That said, I think defining the actual boundaries of the API is probably out of scope for this PR.